test/unit/compiler/passes/merge-character-classes.spec.js

```diff
@@ -0,0 +1,146 @@
+"use strict";
+
+const chai = require("chai");
+const helpers = require("./helpers");
+const pass = require("../../../../lib/compiler/passes/merge-character-classes");
+
+chai.use(helpers);
+
+const expect = chai.expect;
+
+describe("compiler pass |mergeCharacterClasses|", () => {
+  it("Merges single character literals, class, and ref_rules", () => {
+    expect(pass).to.changeAST(
+      [
+        "one = three / $('a' / 'd') / [c-f] / [efh] / [c-h] / [c-g]",
+        "two = 'P' / 'P' / three / 'P' / [Q-T]",
+        "three = $('x' / [u-w])",
+        "four = 'a' / [aaa] / 'a'",
```

Comment:
Because merging also should work when all literal / classes ignores case, I suggest copy the whole test and add `i` suffix here to test `ignoreCase` case